secant/Utils/Zatoshi.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Zatoshi.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 26.05.2022.
+//
+
+import Foundation
+
+typealias ZEC = NSDecimalNumber
+
+// MARK: - Zatoshi
+
+struct Zatoshi {
+    @Clamped(-(100_000_000 * 21_000_000)...(100_000_000 * 21_000_000))
+    var amount: Int64 = 0
+
+    /// Converts `Zatoshi` to `ZEC`
+    var asZec: ZEC {
+        ZEC(value: Double(amount) / Double(100_000_000))
```

Comment:
When doing this conversion, we may want to be doing this as NSDecimals instead of Doubles to avoid potential floating point precision issues.  I'm less familiar with the Swift APIs, although I think that's roughly equivalent to using Java's BigDecimal versus double.

Reply:
refactored